### PR TITLE
Dispose components on death

### LINKED ISSUE #991: Ghost can pick up stuff and the dead player take it in hand.
## Summary

As a ghost, try to pick up stuff, you'll see your item appearing in your dead body hand.

Ghost before picking up an item : 

image

Ghost after picking up an item : 

image

## Steps to Reproduce

in editor, press f12 and type "kill editorUser" then try to pick up an item as a ghost.

## Expected Behavior

No item picking as a ghost.

## Possible solutions

Simply deactivate all scripts on the human prefab upon death, simply make sure that this change is registered over the network. 
It's somewhere in healthcontroller script.

### PR DESCRIPTION
**Pull request by xk0fe**
## Summary
Dispose hands, inventory and interactions also hide inventory for ghost ([reference](https://www.youtube.com/watch?v=45ryS19cNy4))

## Known issues
Issues below were present before the PR
* Player's body is floating above the ground after the rotation is applied to it,
* If the match restarts player can't walk on spawn

## Fixes (optional)
Closes #991

### CODE REVIEW

File: Assets/Scripts/SS3D/Systems/Health/HealthController.cs [hunk 1]
```diff
@@ -23,17 +21,28 @@ private void BecomeGhost(GameObject player, GameObject ghost)
...
+
+    [ObserversRpc]
+    private void RpcDestroyObjects(Entity originEntity)
+    {
+        GameObject originEntityGameObject = originEntity.gameObject;
+        originEntityGameObject.GetComponent<Hands>().Dispose(true);
+        originEntityGameObject.GetComponent<Inventory>().Dispose(true);
+        originEntityGameObject.GetComponent<InteractionController>().Dispose(true);
+        originEntityGameObject.GetComponent<StaminaController>().Dispose(true);
+        originEntityGameObject.GetComponent<HumanoidController>().Dispose(true);
```
Comment (joaoburatto) [inline Assets/Scripts/SS3D/Systems/Health/HealthController.cs:37]: I'm fine with this for now. But please add a comment saying it is a temporary solution and it has to be optimized later on using a `TODO:` tag, like:
`TODO: Optimize these GetComponents, this is a temporary solution.`

File: Assets/Scripts/SS3D/Systems/Health/HealthController.cs [hunk 2]
```diff
@@ -23,17 +21,28 @@ private void BecomeGhost(GameObject player, GameObject ghost)
...
     [ObserversRpc]
-    private void destroyObjects(Entity originEntity, Entity ghostEntity)
+    private void RpcUpdateGhostPosition(Entity originEntity, Entity ghostEntity)
     {
-        originEntity.gameObject.GetComponent<StaminaController>().Dispose(true);
-        originEntity.gameObject.GetComponent<HumanoidController>().Dispose(true);
-        originEntity.gameObject.transform.Rotate(new Vector3(90, 0, 0));
-        ghostEntity.transform.position = originEntity.transform.position;
-        ghostEntity.transform.rotation = originEntity.transform.rotation;
+        Transform originEntityTransform = originEntity.transform;
```
Comment (joaoburatto) [inline Assets/Scripts/SS3D/Systems/Health/HealthController.cs]: You don't need to cache the transform, we already have that inside the NetworkActor class. You can have something like:

```
ghostEntity.Transform.SetPositionAndRotation(originEntityTransform.position, originEntityTransform.rotation);
originEntity.Transform.Rotate(new Vector3(90, 0, 0));
```

File: Assets/Scripts/SS3D/Systems/Interactions/InteractionController.cs [hunk 3]
```diff
@@ -57,12 +57,23 @@ public override void OnStartClient()
         public override void OnStopClient()
         {
             base.OnStopClient();
-            
+            UnsubscribeFromEvents();
+        }
+
+        private void UnsubscribeFromEvents()
+        {
+            if (!Owner.IsLocalClient) return;
```
Comment (joaoburatto) [inline Assets/Scripts/SS3D/Systems/Interactions/InteractionController.cs]: add a space or brackets to not clutter these lines

**Review by joaoburatto — DISMISSED** [review]

**Comment by xk0fe** [comment]
Thanks! I've updated PR according to request

**Review by stilnat — APPROVED** [review]
tested on host and client, working fine, good code, good to go

**Pull request merged**